app/controllers/dashboard/trading_names_controller.rb

```diff
@@ -0,0 +1,52 @@
+module Dashboard
+  class TradingNamesController < ApplicationController
+    before_action :authenticate_user!
+
+    def edit
+      @firm = Firm.find(params[:id])
+      render 'dashboard/firms/edit'
+    end
+
+    def update
+      @firm = Firm.find(params[:id])
+      @firm.update(firm_params) && flash[:notice] = I18n.t('dashboard.firm_edit.saved')
+      render 'dashboard/firms/edit'
+    end
+
+    private
+
+    def principal
+      current_user.principal
+    end
+
+    FIRM_PARAMS = [
+      :email_address,
+      :telephone_number,
+      :address_line_one,
+      :address_line_two,
+      :address_town,
+      :address_county,
+      :address_postcode,
+      :free_initial_meeting,
+      :initial_meeting_duration_id,
+      :minimum_fixed_fee,
+      :retirement_income_products_percent,
+      :pension_transfer_percent,
+      :long_term_care_percent,
+      :equity_release_percent,
+      :inheritance_tax_and_estate_planning_percent,
+      :wills_and_probate_percent,
+      :other_percent,
+      in_person_advice_method_ids: [],
+      other_advice_method_ids: [],
+      initial_advice_fee_structure_ids: [],
+      ongoing_advice_fee_structure_ids: [],
+      allowed_payment_method_ids: [],
+      investment_size_ids: []
+    ]
```

Comment:
```
irb(main):001:0> class Thing
irb(main):002:1> private
irb(main):003:1> STUFF = [0, 1]
irb(main):004:1> end
=> [0, 1]
irb(main):005:0> Thing::STUFF
=> [0, 1]
irb(main):006:0> Thing::STUFF.pop
=> 1
irb(main):007:0> Thing::STUFF
=> [0]
```

Groan. You are right, private or not makes no difference when what we want is `FIRM_PARAMS = [...].freeze`. Good Ruby-fu there sir.